### PR TITLE
DROOLS-6630 DMN Signavio Profile multi-instance with complex inputs

### PR DESCRIPTION
**Pull request by tarilabs**
**JIRA**: https://issues.redhat.com/browse/DROOLS-6630

**referenced Pull Requests**: none

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>a pull request</b> please add comment: <b>Jenkins retest this</b>
 
* <b>a full downstream build</b> please add comment: <b>Jenkins run fdb</b>
  
* <b>a compile downstream build</b> please  add comment: <b>Jenkins run cdb</b>

* <b>a full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>

* <b>an upstream build</b> please add comment: <b>Jenkins run upstream</b>
</details>

### CODE REVIEW

**Review by hellowdan — APPROVED** [review]

File: kie-dmn/kie-dmn-signavio/src/main/java/org/kie/dmn/signavio/MultiInstanceDecisionLogic.java [hunk 1]
```diff
@@ -163,7 +166,7 @@ public EvaluatorResult evaluate(DMNRuntimeEventManager eventManager, DMNResult d
             List<? super Object> invokationResults = new ArrayList<>();
             
             try {
-                Object cycleOnRaw = dmnContext.get( mi.iterationExpression );
+                Object cycleOnRaw = feel.evaluate(mi.iterationExpression, dmnContext.getAll());
```
Comment (j-beyer) [inline kie-dmn/kie-dmn-signavio/src/main/java/org/kie/dmn/signavio/MultiInstanceDecisionLogic.java:169]: You mentioned the assumption:
>In this case, I believe I achieved a fix understanding that the <sigExt:iterationExpression> is a generic FEEL expression valid in the original DMN context scope.

I guess that's fair. From my understanding, we serialize this iteration expression from a "reference expression", i.e. it probably won't be any generic FEEL expression, rather just always a reference to another variable (not sure if that's the correct DMN term 😅).

But still I believe that resolving this via FEEL as you did here should lead to the desired results :)

**Review by j-beyer — APPROVED** [review]
Looks good to me. My colleague will also run a manual test, and we'll get back to you once we're finished 🙂

**Comment by tarilabs** [comment]
> https://github.com/kiegroup/drools/pull/3876#pullrequestreview-772818806

thank you @j-beyer , glad to hear this initial positive feedback, and will await your team final confirmation 👍

**Review by pibizza — APPROVED** [review]

**Comment by tarilabs** [comment]
Jenkins retest this
Jenkins run cdb

**Comment by j-beyer** [comment]
🚀 Tests look good, we should be good to merge. Again, thanks a lot!
Could you tell us the release version where this will be included, and when that release will be published?

During our tests, we've uncovered another potential issue with MIDs. We're still investigating where this would need to be fixed (i.e., maybe already on our side in the export). Just warning you that we might reach out again in the next days 😅

**Comment by tarilabs** [comment]
```
12:56:00  [INFO] --- maven-javadoc-plugin:3.0.1:jar (default) @ drools-serialization-protobuf ---
12:56:20  Cannot contact kie-psi-rhel7-large-181: java.lang.InterruptedException
15:22:18  Cancelling nested steps due to timeout
15:22:18  Could not connect to kie-psi-rhel7-large-181 to send interrupt signal to process
[Pipeline] }
15:22:18  kie-psi-rhel7-large-181 was marked offline: Connection was broken: java.io.EOFException
```

🤷 

Jenkins retest this

**Comment by tarilabs** [comment]
@j-beyer 

> Could you tell us the release version where this will be included, and when that release will be published?

I will need to create a subsequent PR targeting `7.x` branch, once this is eventually merged. When the other PR (will be linked here) will be merged, we will know for certain the version including this fix 👍

**Comment by tarilabs** [comment]
When https://github.com/kiegroup/drools/commit/c6b1cbf07f324bd722723d08178ca4367dd36208 will belong to a 7.x release the tag should appear, I would expect `7.61.0.Final`

**Pull request merged**